File: Continuous Subarray Sum.cpp
class Solution {
public:
    /**
     * @param A an integer array
     * @return  A list of integers includes the index of 
     *          the first number and the index of the last number
     */
    vector<int> continuousSubarraySum(vector<int>& A) {
        // Write your code here
        int sum = 0;
        int left = 0, right = 0, pre_left = 0;
        int max = INT_MIN;
        for (int i = 0; i < A.size(); ++i)
        {
            sum += A[i];
            if (max < sum)
            {
                max = sum;
                right = i;
                left = pre_left;
            }
            if (sum < 0)
            {
                sum = 0;
                pre_left = i + 1;
            }
        }
        return {left, right};
    }
};

File: Continuous Subarray Sum.py
class Solution:
    # @param {int[]} A an integer array
    # @return {int[]}  A list of integers includes the index of the 
    #                  first number and the index of the last number
    def continuousSubarraySum(self, A):
        # Write your code here
        if not A:
            return []
        length = len(A)
        ans = [-1, -1]
        pre = -1
        local, gl = float('-inf'), float('-inf')
        for i in range(length):
            if local + A[i] < A[i]:
                local = A[i]
                pre = i
            else:
                local += A[i]
            if gl < local:
                gl = local
                ans[0] = pre
                ans[1] = i
        return ans
        